cmake/modules/CheckCompiler.cmake

```diff
@@ -97,43 +97,31 @@ endif()
 include(CheckCXXCompilerFlag)
 include(CheckCCompilerFlag)
 
-#---Check for cxx11 option------------------------------------------------------------
-if(cxx11 AND cxx14)
-  message(STATUS "c++11 mode requested but superseded by request for c++14 mode")
-  set(cxx11 OFF CACHE BOOL "" FORCE)
-endif()
-if((cxx11 OR cxx14) AND cxx17)
-  message(STATUS "c++11 or c++14 mode requested but superseded by request for c++17 mode")
-  set(cxx11 OFF CACHE BOOL "" FORCE)
-  set(cxx14 OFF CACHE BOOL "" FORCE)
-endif()
-if(cxx11)
-  CHECK_CXX_COMPILER_FLAG("-std=c++11" HAS_CXX11)
-  if(NOT HAS_CXX11)
-    message(STATUS "Current compiler does not suppport -std=c++11 option. Switching OFF cxx11 option")
-    set(cxx11 OFF CACHE BOOL "" FORCE)
-  endif()
-endif()
-if(cxx14)
-  CHECK_CXX_COMPILER_FLAG("-std=c++14" HAS_CXX14)
-  if(NOT HAS_CXX14)
-    message(STATUS "Current compiler does not suppport -std=c++14 option. Switching OFF cxx14 option")
-    set(cxx14 OFF CACHE BOOL "" FORCE)
-  endif()
+#---C++ standard----------------------------------------------------------------------
+
+if(cxx11 OR cxx14 OR cxx17)
+  message(DEPRECATION "Options cxx11/14/17 are deprecated. Please use CMAKE_CXX_STANDARD instead.")
 endif()
-if(cxx17)
-  CHECK_CXX_COMPILER_FLAG("-std=c++17" HAS_CXX17)
-  if(NOT HAS_CXX17)
-    message(STATUS "Current compiler does not suppport -std=c++17 option. Switching OFF cxx17 option")
-    set(cxx17 OFF CACHE BOOL "" FORCE)
+
+if(NOT DEFINED CMAKE_CXX_STANDARD)
+  # for backward compatibility
+  if(cxx17)
+    set(CMAKE_CXX_STANDARD 17)
+  elseif(cxx14)
+    set(CMAKE_CXX_STANDARD 14)
+  else()
+    set(CMAKE_CXX_STANDARD 11)
   endif()
 endif()
-if(root7)
-  if(cxx11)
-    message(STATUS "ROOT7 interfaces require >= cxx14 which is disabled. Switching OFF root7 option")
-    set(root7 OFF CACHE BOOL "" FORCE)
-  endif()
-    set(http ON CACHE BOOL "" FORCE)
+
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+set(CMAKE_CXX_EXTENSIONS FALSE)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++${CMAKE_CXX_STANDARD}")
+
+if(root7 AND (NOT http OR CMAKE_CXX_STANDARD EQUAL 11))
```

Comment:
That does not look equivalent to the original logic. I.e. `-Droot7=On -Dcxx11=On` would trigger the warning, whatever the setting of `http`, and with `root7` enabled, `http` would be `On`.

Reply:
I can make it more or less equivalent to before, like this:
```cmake
if(root7)
  if(CMAKE_CXX_STANDARD EQUAL 11)
    message(WARNING "ROOT 7 requires C++14 or higher. Switching OFF root7 option.")
  else()
    set(http ON CACHE BOOL "" FORCE)
  endif()
endif()
```
Note that `http` is turned ON regardless in the current code.